### PR TITLE
Address some nuget package warnings

### PR DESCRIPTION
**Pull request by jviau**
### Pull request checklist

**IMPORTANT**: Currently, changes must be backported to the `in-proc` branch to be included in Core Tools and non-Flex deployments.

* [x] My changes **do not** require documentation changes
    * [ ] Otherwise: Documentation issue linked to PR
* [x] My changes **should not** be added to the release notes for the next release
    * [ ] Otherwise: I've added my notes to `release_notes.md`
* [x] My changes **do not** need to be backported to a previous version
    * [ ] Otherwise: Backport tracked by issue/PR #issue_or_pr
* [x] My changes **do not** require diagnostic events changes
    * Otherwise: I have added/updated all related diagnostic events and their documentation (Documentation issue linked to PR)
* [x] I have added all required tests (Unit tests, E2E tests)

### Additional information

Fixes an issue with nuget package audits and some test dependencies.

### CODE REVIEW

File: test/WebJobs.Script.Tests.Integration/WebJobs.Script.Tests.Integration.csproj [hunk 1]
```diff
@@ -37,23 +37,25 @@
...
     <PackageReference Include="xunit" Version="2.4.1" />
-    <PackageReference Include="xunit.runner.visualstudio" Version="2.4.3">
-      <PrivateAssets>all</PrivateAssets>
-      <IncludeAssets>runtime; build; native; contentfiles; analyzers</IncludeAssets>
-    </PackageReference>
+    <PackageReference Include="xunit.runner.visualstudio" Version="2.4.3" PrivateAssets="all" />
+  </ItemGroup>
+
+  <ItemGroup Condition="'$(TargetFramework)' == 'net6.0'">
+    <PackageReference Include="Microsoft.Extensions.Logging.Console" Version="8.0.1" />
```
Comment (surgupta-msft) [inline test/WebJobs.Script.Tests.Integration/WebJobs.Script.Tests.Integration.csproj]: Shouldn't this be version 6.0.1?

File: test/WebJobs.Script.Tests.Integration/WebJobs.Script.Tests.Integration.csproj [hunk 2]
```diff
@@ -37,23 +37,25 @@
...
+  </ItemGroup>
+
+  <ItemGroup Condition="'$(TargetFramework)' == 'net8.0'">
+    <PackageReference Include="Microsoft.Extensions.Logging.Console" Version="6.0.1" />
```
Comment (surgupta-msft) [inline test/WebJobs.Script.Tests.Integration/WebJobs.Script.Tests.Integration.csproj]: Do you mean version 8.0.1?

File: test/WebJobs.Script.Tests.Integration/WebJobs.Script.Tests.Integration.csproj [hunk 3]
```diff
@@ -37,23 +37,25 @@
...
-      <NoWarn>NU1701</NoWarn>
-    </PackageReference>
+    <PackageReference Include="Microsoft.AspNet.WebApi.Core" Version="5.2.6" NoWarn="NU1701" />
     <PackageReference Include="Microsoft.AspNetCore.TestHost" Version="6.0.0" />
     <PackageReference Include="Microsoft.Azure.DocumentDB.Core" Version="2.11.2" />
     <PackageReference Include="Microsoft.Azure.Functions.NodeJsWorker" Version="3.10.1" />
     <PackageReference Include="Microsoft.Azure.WebJobs.Extensions.Storage" Version="4.0.5-11874" />
     <PackageReference Include="Microsoft.Azure.Functions.JavaWorker" Version="2.17.0" />
     <PackageReference Include="Microsoft.Azure.Mobile.Client" Version="4.0.2" />
-    <PackageReference Include="Microsoft.Extensions.Logging.Console" Version="8.0.0" />
```
Comment (surgupta-msft) [inline test/WebJobs.Script.Tests.Integration/WebJobs.Script.Tests.Integration.csproj:49]: Are we getting any warnings for using this package for both net6 and net8?

File: test/WebJobs.Script.Tests.Integration/WebJobs.Script.Tests.Integration.csproj (hunk 1 above)
Comment (jviau) [reply test/WebJobs.Script.Tests.Integration/WebJobs.Script.Tests.Integration.csproj]: Yes, didn't realize I reversed these.

File: test/WebJobs.Script.Tests.Integration/WebJobs.Script.Tests.Integration.csproj (hunk 3 above)
Comment (jviau) [reply test/WebJobs.Script.Tests.Integration/WebJobs.Script.Tests.Integration.csproj:49]: It was two things: Using `8.0.0` brought in a vulnerable version of `System.Text.Json`, which was producing warnings. And the other is that by using `8.x` in the net6 TFM, we were uplifting a bunch of transitive dependencies to 8.x, when we really should be testing with 6.x (what the host uses)

**Review by surgupta-msft — APPROVED** [review]

**Review by mathewc — APPROVED** [review]

**Pull request merged**